### ISSUE TITLE
Not able to upload attachment to chatlines and getting 400 error code on sending rich-text content in message.

### ISSUE THREAD

**Issue by loogitakurrey** [basecamp/bc3-api#240]
1. I am trying to upload an attachment in the Chat Lines but it is not getting created. 
`{
  "content": "Good morning"
} `
This is a simple text but I want to upload attachments also. How to do that?
2. Another thing I am trying is the rich text bg-attachment HTML text inside content in the message. But it is giving 400 error.
`{
  "subject": "Kickoff",
  "content": "<div>testing</div><bc-attachment sgid="BAh7CEkiCGdpZAY6BkVUSSIuZ2lkOi8vYmMzL0F0dGFjaG1lbnQvMTM4MzAxMjEwP2V4cGlyZXNfaW4GOwBUSSIMcHVycG9zZQY7AFRJIg9hdHRhY2hhYmxlBjsAVEkiD2V4cGlyZXNfYXQGOwBUMA==--539659284a0a6790af7c6bc8528c79ddf85e6035" Content-Type="image/jpeg" caption="My photo"></bc-attachment>",
  "status": "active"
}`
Please let me know what I am doing wrong in the above json.
I am using basecamp 3 api.
Thank you.

**Comment by rosa** [comment]
Hey @loogitakurrey, did you see my comment in the other issue you opened? https://github.com/basecamp/bc3-api/issues/239#issuecomment-527114443
Have you tried using Chatbots?

**Comment by loogitakurrey** [comment]
Thank you for your reply. I will check it. Do you have any idea about the 2nd question?

**Comment by rosa** [comment]
Ah, my apologies! I didn't realise your second question was about a message. Does the `sgid` correspond to a newly created attachment via the [Create attachment endpoint](https://github.com/basecamp/bc3-api/blob/master/sections/attachments.md#create-an-attachment), or has it been reused? Do you have the value for `X-Request-Id` returned in the 400 response? Might be handy to check our logs and figure out what's wrong.

**Comment by loogitakurrey** [comment]
Yes, I am using the same API to create the attachment. and using that into the content of the message inside the `<bc-attachment sgid=''></bc-attachment>` . 
My request Id is  [X-Request-Id] => 80f2c442-d90d-4f02-83f2-ed876262acb5 
Please let me know what is the issue.
Below I am sending the whole curl request I am making:
`$data='{
  "subject": "Kickoff",
  "content": "<div><bc-attachment sgid="BAh7CEkiCGdpZAY6BkVUSSIuZ2lkOi8vYmMzL0F0dGFjaG1lbnQvMTM5NjU1NjAxP2V4cGlyZXNfaW4GOwBUSSIMcHVycG9zZQY7AFRJIg9hdHRhY2hhYmxlBjsAVEkiD2V4cGly..............."></bc-attachment></div>",
  "status": "active"
}';
$account_id='467865';
$access_token='BAhbB0kiAbB7ImNsaWVudF9pZCI6IjRhYzMxNzMwMzI1MjMwNThiZGE3YjUzZDE0MjU4ZGMxNDlkMGIyMjEiLCJleHBpcmVzX2F0IjoiMjAxOS0wOS0xMlQwOTowNDowN1oiLCJ1c2VyX2lkcyI6WzQwMTkwMjc4XSwidmVyc2lvbiI6MSwiYXBpX2RlYWRib2x0IjoiOThhYzM4NGU0ZDljMTEwMjg5YTUxMTk4NjY4YWFmY......................';
$ch = curl_init();

curl_setopt($ch, CURLOPT_URL, "https://3.basecampapi.com/$account_id/buckets/$bucket_id/message_boards/$message_board_id/messages.json");
curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($ch, CURLOPT_POSTFIELDS, $data);
curl_setopt($ch, CURLOPT_POST, 1);

$headers = array();
$headers[] = 'Authorization: Bearer '.$access_token;
$headers[] = 'Content-Type: application/json';
$headers[] = 'Content-Length: '.strlen($data);
$headers[] = 'User-Agent: BasecampGDS (dileep.constacloud@gmail.com)';
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
curl_setopt($ch, CURLOPT_HEADER, 1);
$result = curl_exec($ch);
if (curl_errno($ch)) {
    echo 'Error:' . curl_error($ch);
}
curl_close($ch);`

**Comment by rosa** [comment]
@loogitakurrey, looks like the JSON you're sending as the value for `$data` is malformed and we can't parse it, I think the problem might be the `"` extra you're using unescaped for the `sgid` value. 

About the other issue:
> I am trying to upload an attachment in the Chat Lines but it is not getting created.

The reply I gave you in the other issue (https://github.com/basecamp/bc3-api/issues/239#issuecomment-527114443) is incorrect and our docs aren't very clear, you can't add rich text content to chat lines, and can't add attachments to chat lines from the API. I'll be clarifying this in that other issue and in the docs.

**Comment by rosa** [comment]
> The reply I gave you in the other issue (#239 (comment)) is incorrect and our docs aren't very clear, you can't add rich text content to chat lines, and can't add attachments to chat lines from the API. I'll be clarifying this in that other issue and in the docs.

Oops, sorry! I was right the first time 😅 Got confused with something else and thought I wasn't. That comment in the other reply is still valid, you can add rich text and attachments to chat lines via the API but only from chatbots. When you do that, consider also the issue above with badly formed JSON.

**Comment by loogitakurrey** [comment]
Thank you for your reply. Creating attachment in chat line is possible through UI and not through API Isn't?

**Comment by rosa** [comment]
Ah, yes, that's right @loogitakurrey. The only way is via chat bots and in that way, the attachment will be always created by the chat bot.

**Comment by rosa** [comment]
Closing this one, since I don't think there's anything pending 🙂
